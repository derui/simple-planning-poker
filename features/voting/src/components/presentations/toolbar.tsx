import { Variant } from "@spp/shared-color-variant";
import { Icon, Icons } from "@spp/ui-icon";
import { ToggleButton } from "@spp/ui-toggle-button";
import { clsx } from "clsx";

export interface Props {
  /**
   * Event to notify change role between player and inspector.
   */
  onChangeRole?: (role: "player" | "inspector") => void;

  /**
   * Default role
   */
  defaultRole: "player" | "inspector";
}

const styles = {
  root: clsx("grid", "grid-rows-1", "grid-cols-1", "place-content-center", "gap-2", "place-items-center"),
  role: clsx(
    "grid",
    "grid-rows-1",
    "grid-cols-[auto_auto_auto]",
    "rounded",
    "bg-orange-100",
    "place-content-center",
    "gap-2",
    "place-items-center",
    "px-2",
    "py-1"
  ),
} as const;

export const Toolbar = function Toolbar({ onChangeRole, defaultRole }: Props) {
  const handleToggle = (checked: boolean) => {
    if (checked) {
      onChangeRole?.("inspector");
    } else {
      onChangeRole?.("player");
    }
  };

  return (
    <div className={styles.root}>
      <div className={styles.role}>
        <Icon type={Icons.user} variant={Variant.orange} />
        <ToggleButton initialChecked={defaultRole == "inspector"} onToggle={handleToggle} />
        <Icon type={Icons.eye} variant={Variant.orange} />
      </div>
    </div>
  );
};
