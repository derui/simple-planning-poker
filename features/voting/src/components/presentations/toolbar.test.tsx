import { cleanup, render, screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import { afterEach, expect, test } from "vitest";
import { Toolbar } from "./toolbar.js";

afterEach(cleanup);

test("should be able to render", () => {
  const ret = render(<Toolbar defaultRole="player" />);

  expect(ret.container).toMatchSnapshot();
});

test("should not call handler after initial render", () => {
  render(
    <Toolbar
      defaultRole="inspector"
      onToggleRole={() => {
        expect.fail("Do not call initial render");
      }}
    />
  );
});

test("get changed role from player to inspector", async () => {
  expect.assertions(1);

  render(
    <Toolbar
      defaultRole="player"
      onToggleRole={() => {
        expect(true).toBe(true);
      }}
    />
  );

  await userEvent.click(screen.getByText("Player"));
});

test("get changed role from inspector to player", async () => {
  expect.assertions(1);

  render(
    <Toolbar
      defaultRole="inspector"
      onToggleRole={() => {
        expect(true).toBe(true);
      }}
    />
  );

  await userEvent.click(screen.getByText("Inspector"));
});
