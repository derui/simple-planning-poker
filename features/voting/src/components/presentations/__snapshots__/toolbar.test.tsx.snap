// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should be able to render 1`] = `
<div>
  <div
    class="grid grid-rows-1 grid-cols-1 place-content-center gap-2 place-items-center"
  >
    <div
      class="grid grid-rows-1 grid-cols-[auto_auto_auto] rounded bg-orange-100 place-content-center gap-2 place-items-center px-2 py-1"
    >
      <span
        class="flex before:bg-orange-700 before:inline-block before:flex-none before:[mask-image:url(/static/svg/tabler-icons/user.svg)] before:[mask-size:cover] before:[mask-position:center] before:[mask-repeat:no-repeat] before:w-6 before:h-6 w-6 h-6"
      />
      <span
        class="flex flex-none outline-none border-none p-0 h-8 items-center rounded"
        role="switch"
      >
        <span
          class="flex relative inline-block bg-transparent h-6 justify-items-center border border-emerald-600 rounded"
        >
          <span
            class="relative inline-block m-0 px-1 w-12 h-full transition-colors overflow-hidden rounded bg-emerald-100"
          >
            <span
              class="absolute inline-block bg-emerald-500 h-full w-6 left-0 top-0 transition-transform"
            />
          </span>
          <input
            class="hidden"
            type="checkbox"
          />
        </span>
      </span>
      <span
        class="flex before:bg-orange-700 before:inline-block before:flex-none before:[mask-image:url(/static/svg/tabler-icons/eye.svg)] before:[mask-size:cover] before:[mask-position:center] before:[mask-repeat:no-repeat] before:w-6 before:h-6 w-6 h-6"
      />
    </div>
  </div>
</div>
`;

exports[`should change current check 1`] = `
<div>
  <div
    class="grid grid-rows-1 grid-cols-1 place-content-center gap-2 place-items-center"
  >
    <div
      class="grid grid-rows-1 grid-cols-[auto_auto_auto] rounded bg-orange-100 place-content-center gap-2 place-items-center px-2 py-1"
    >
      <span
        class="flex before:bg-orange-700 before:inline-block before:flex-none before:[mask-image:url(/static/svg/tabler-icons/user.svg)] before:[mask-size:cover] before:[mask-position:center] before:[mask-repeat:no-repeat] before:w-6 before:h-6 w-6 h-6"
      />
      <span
        class="flex flex-none outline-none border-none p-0 h-8 items-center rounded"
        role="switch"
      >
        <span
          class="flex relative inline-block bg-transparent h-6 justify-items-center border border-emerald-600 rounded"
        >
          <span
            class="relative inline-block m-0 px-1 w-12 h-full transition-colors overflow-hidden rounded bg-emerald-300"
          >
            <span
              class="absolute inline-block bg-emerald-500 h-full w-6 left-0 top-0 transition-transform translate-x-6"
            />
          </span>
          <input
            checked=""
            class="hidden"
            type="checkbox"
          />
        </span>
      </span>
      <span
        class="flex before:bg-orange-700 before:inline-block before:flex-none before:[mask-image:url(/static/svg/tabler-icons/eye.svg)] before:[mask-size:cover] before:[mask-position:center] before:[mask-repeat:no-repeat] before:w-6 before:h-6 w-6 h-6"
      />
    </div>
  </div>
</div>
`;
